pkg/microservice/aslan/core/multicluster/service/cache.go

```diff
@@ -0,0 +1,111 @@
+limitations under the License.
+*/
+
+package service
+
+import (
+	"context"
+	"fmt"
+
+	corev1 "k8s.io/api/core/v1"
+	storagev1 "k8s.io/api/storage/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	"github.com/koderover/zadig/pkg/microservice/aslan/config"
+	"github.com/koderover/zadig/pkg/setting"
+	kubeclient "github.com/koderover/zadig/pkg/shared/kube/client"
+	"github.com/koderover/zadig/pkg/tool/log"
+)
+
+type PVC struct {
+	Name               string `json:"name"`
+	StorageSizeInBytes int64  `json:"storage_size_in_bytes"`
+}
+
+func ListStorageClasses(ctx context.Context, clusterID string) ([]string, error) {
+	kclient, err := kubeclient.GetKubeClient(config.HubServerAddress(), clusterID)
+	if err != nil {
+		return nil, fmt.Errorf("failed to get kube client: %s", err)
+	}
+
+	var namespace string
+	switch clusterID {
+	case setting.LocalClusterID:
+		// For cluster-level resources, we need to explicitly configure the namespace to be empty.
+		namespace = ""
+	default:
+		if namespace != AttachedClusterNamespace {
+			return nil, fmt.Errorf("invalid namespace in attached cluster: %s. Valid: %s", namespace, AttachedClusterNamespace)
+		}
+	}
+
+	scList := &storagev1.StorageClassList{}
+	err = kclient.List(ctx, scList, client.InNamespace(namespace))
+	if err != nil {
+		return nil, fmt.Errorf("failed to list storageclasses: %s", err)
+	}
+	log.Infof("StorageClass Length: %v", len(scList.Items))
+
+	storageclasses := make([]string, 0, len(scList.Items))
+	for _, item := range scList.Items {
+		if !StorageProvisioner(item.Provisioner).IsNFS() {
+			continue
+		}
+
+		storageclasses = append(storageclasses, item.Name)
+	}
+
+	return storageclasses, nil
+}
+
+func ListPVCs(ctx context.Context, clusterID, namespace string) ([]PVC, error) {
+	kclient, err := kubeclient.GetKubeClient(config.HubServerAddress(), clusterID)
+	if err != nil {
+		return nil, fmt.Errorf("failed to get kube client: %s", err)
+	}
+
+	switch clusterID {
+	case setting.LocalClusterID:
+		// For now caller may not know exactly which namespace Zadig is deployed to, so a correction is made here.
+		namespace = config.Namespace()
+	default:
+		if namespace != AttachedClusterNamespace {
+			return nil, fmt.Errorf("invalid namespace in attached cluster: %s. Valid: %s", namespace, AttachedClusterNamespace)
+		}
+	}
+
+	pvcList := &corev1.PersistentVolumeClaimList{}
+	err = kclient.List(ctx, pvcList, client.InNamespace(namespace))
+	if err != nil {
+		return nil, err
+	}
+	log.Infof("PVC Length: %d", len(pvcList.Items))
+
+	pvcs := make([]PVC, 0, len(pvcList.Items))
+	for _, item := range pvcList.Items {
+		// Don't support `corev1.PersistentVolumeBlock` because the PVC would be mounted on multiple nodes.
+		if item.Spec.VolumeMode != nil && *item.Spec.VolumeMode == corev1.PersistentVolumeBlock {
+			continue
+		}
+
+		pvcs = append(pvcs, PVC{
+			Name:               item.Name,
+			StorageSizeInBytes: item.Spec.Resources.Requests.Storage().Value(),
```

Comment:
which may be empty？

Reply:
When creating a PVC, `requests` are required, so it will not be empty.